### PR TITLE
Add configurable base controller class

### PR DESCRIPTION
**Pull request by elalemanyo**
Add `base_controller_class` accessor to ActiveAnalytics. I think this may help when using the mountable dashboard.

### CODE REVIEW

File: lib/active_analytics.rb [hunk 1]
```diff
@@ -2,6 +2,8 @@
 require "active_analytics/engine"
 
 module ActiveAnalytics
+  mattr_accessor :base_controller_class
```
Comment (alexisbernard) [inline lib/active_analytics.rb]: It should have a default value `ActionController::Base`.

File: README.md [hunk 2]
```diff
@@ -38,6 +38,15 @@ Add the route to ActiveAnalytics dashboard at the desired endpoint:
 # config/routes.rb
 mount ActiveAnalytics::Engine, at: "analytics"  # http://localhost:3000/analytics
 ```
+By default ActiveAnalytics will extend `ActionController::Base`, but you can specify a custom base controller for the ActiveAnalytics dashboard:
+
+```ruby
+# config/initializers/active_analytics.rb
+Rails.application.configure do
+  ActiveAnalytics.base_controller_class = "ApplicationController"
```
Comment (alexisbernard) [inline README.md]: Probably AdminController is a better example, since analytics are private for most apps.

Comment (elalemanyo) [inline README.md]:
```suggestion
  ActiveAnalytics.base_controller_class = "AdminController"
```

File: lib/active_analytics.rb (hunk 1 above)
Comment (elalemanyo) [inline lib/active_analytics.rb]:
```suggestion
  mattr_accessor :base_controller_class, default: "ActionController::Base"
```

**Comment by alexisbernard** [comment]
Thanks!

**Pull request merged**